### ISSUE TITLE
Map colors for "outfit in storage" and "has an outfitter" are hard to distinguish

### ISSUE THREAD

**Issue by hexagonrecursion** [endless-sky/endless-sky#9400]
### Problem Description

It's nice that I can store outfits on a planet. It's also nice that the map reminds me which systems I left something in:

![image](https://github.com/endless-sky/endless-sky/assets/52621858/5815a694-0610-4444-a232-d088b96673ea)

The problem: the map has a mode that can be used to find where I can get a specific outfit. It highlights the systems where it can be bought. It also highlights systems where I stored this outfit. The yellowish indicator for "sells this outfit" is clearly visible, but the greenish indicator for "outfit in storage" is almost indistinguishable from the blueish indicator for "has an outfitter":

![image](https://github.com/endless-sky/endless-sky/assets/52621858/cf2082cc-d566-45ff-8ce2-047eca9b94ed)

### Related Issue Links

none?

### Desired Solution

Change the color for "outfit in storage" to be more distinct

### Alternative Approaches

Don't highlight systems that merely have an outfitter when the map is in the "outfitters" mode and a specific outfit is selected. Then there is nothing to confuse the greenish indicator with

**Comment by Amazinite** [comment]
These colors are hard coded at the moment. The MapColor function takes in a double that's clamped to the range [-1, 1] and maps that to part of a range of colors.
https://github.com/endless-sky/endless-sky/blob/8091337b6630c4f041a4498094cf087ed758c367/source/MapPanel.cpp#L656-L674

The value for no outfitter is -1, an outfitter that doesn't sell the selected outfit is 0, an outfitter that does have the selected outfit is 1, and an outfit being in storage is 0.5.

Looks like the reason that a color range is used instead of having discretely defined colors is because the same function is used for the outfitter, shipyard, outfit sales, ship sales, trade, and visited planets map keys. All these map keys actually either use much of the range of values (such as the trade key) or only use two or three colors from the range (such as the visited planets key).

Prior to the addition of this new state on the sales key, there were only three states that this map key could be in, that being a value of -1, 0, and 1, which are relatively distinct colors from one another. The value of 0.5 for the new state is going to naturally lend itself to not being as easy to distinguish, as the intention behind the MapColors function was to have similar input values result in similar output colors.

It makes sense to me that the trade key and other keys that can use more of the range of values would continue to use this color scale, and the other map keys using only two or three states from the scale means they could still use it, but I don't think it makes sense to keep using the color range when we get into the territory of 4+ discrete states, as is now the case with the ship and outfit sales.

We should probably have MapSalesPanel override the MapColor function that it inherits from MapPanel so that it outputs defined colors instead of using a range of colors.

**Comment by mark-c** [comment]
Is there a way to "inject" this or overrule it from any 'external methods' or such?

**Comment by eebop** [comment]
Not really. Maybe we should put all those attributes into a config file?
